lms/djangoapps/bulk_email/tasks.py

```diff
@@ -97,25 +97,31 @@
 )
 
 
-def _get_course_email_context(course):
+def _get_course_email_context(course, is_secure):
     """
     Returns context arguments to apply to all emails, independent of recipient.
+
+    Inputs are:
+    * `course`: Course related to the current email
+    * `is_secure`: Set to True to return https URLs, False to use http.
     """
     course_id = course.id.to_deprecated_string()
     course_title = course.display_name
     course_end_date = get_default_time_display(course.end)
-    course_url = 'https://{}{}'.format(
-        settings.SITE_NAME,
+    scheme = u'https' if is_secure else u'http'
```

Comment:
@pomegranited I don't think `settings.HTTPS` is always True; it's just not auto-detected (by necessity). I think that if you add `HTTPS: false` or similar to `EDXAPP_ENV_EXTRA` in the ansible vars, it will add that to the LMS (and CMS) `.env.json` file, which should be picked up [in lms/envs/aws.py](https://github.com/edx/edx-platform/blob/68312bd/lms/envs/aws.py#L170).

I'm not sure if you can rely on `request.is_secure()` for Open edX deployments, since the SSL is usually handled by a reverse proxy and django itself gets only insecure HTTP requests. So I think you should use `settings.HTTP` instead.